File: RS2-FrizerskiSalon/Database/Uposlenik.cs
using System;
using System.Collections.Generic;

namespace RS2_FrizerskiSalon.Database;

public partial class Uposlenik
{
    #nullable enable
    public int UposlenikId { get; set; }

    public int? ZanimanjeId { get; set; }

    public int? SpolId { get; set; }

    public string? Ime { get; set; }

    public string? Prezime { get; set; }

    public string? Email { get; set; }

    public string? Telefon { get; set; }

    public string? KorisnickoIme { get; set; }

    public string? LozinkaHash { get; set; }

    public string? LozinkaSalt { get; set; }

    public byte[]? Slika { get; set; }

    public virtual ICollection<Izvjestaj> Izvjestajs { get; set; } = new List<Izvjestaj>();

    public virtual ICollection<Obavijest> Obavijests { get; set; } = new List<Obavijest>();

    public virtual Portfolio? Portfolio { get; set; }

    public virtual Spol? Spol { get; set; }

    public virtual ICollection<Termin> Termins { get; set; } = new List<Termin>();

    public virtual Zanimanje? Zanimanje { get; set; }
}

File: FrizerskiSalon.Desktop/frizerski_salon_admin/lib/model/Uposlenik.dart
import 'dart:convert';
import 'Spolovi.dart';
import 'Zanimanja.dart';

class Uposlenici {
  int? uposlenikId;
  String? ime;
  String? prezime;
  String? email;
  String? telefon;
  String? korisnickoIme;
  List<int>? slika;
  int? spolId;
  int? zanimanjeId;
  Spolovi? spol;
  Zanimanja? zanimanje;

  Uposlenici({
    this.uposlenikId,
    this.ime,
    this.prezime,
    this.email,
    this.telefon,
    this.korisnickoIme,
    this.slika,
    this.spolId,
    this.zanimanjeId,
    this.spol,
    this.zanimanje,
  });

  factory Uposlenici.fromJson(Map<String, dynamic> json) {
    String stringByte = json['slika'] as String;
    List<int> bytes = base64.decode(stringByte);

    return Uposlenici(
      uposlenikId: json['uposlenikId'] as int?,
      ime: json['ime'] as String?,
      prezime: json['prezime'] as String?,
      email: json['email'] as String?,
      telefon: json['telefon'] as String?,
      korisnickoIme: json['korisnickoIme'] as String?,
      slika: bytes,
      spolId: json['spolId'] as int?,
      zanimanjeId: json['zanimanjeId'] as int?,
      spol: json['spol'] != null ? Spolovi.fromJson(json['spol']) : null,
    zanimanje: json['zanimanje'] != null ? Zanimanja.fromJson(json['zanimanje']) : null,

    );
  }

  Map<String, dynamic> toJson() => {
        "uposlenikId": uposlenikId,
        "ime": ime,
        "prezime": prezime,
        "email": email,
        "telefon": telefon,
        "korisnickoIme": korisnickoIme,
        "slika": slika,
        "spolId": spolId,
        "zanimanjeId": zanimanjeId,
        "spol": spol?.toJson(),
        "zanimanje": zanimanje?.toJson(),
      };
}
